arcgis-ios-sdk-samples/Edit data/Edit features with feature-linked annotation/EditFeaturesWithFeatureLinkedAnnotationViewController.swift

```diff
@@ -0,0 +1,229 @@
+        })
+        // Present the alert.
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+    // Move the currently selected point feature to the given map point, by updating the selected feature's geometry and feature table.
+    func movePoint(mapPoint: AGSPoint) {
+        guard let selectedFeature = self.selectedFeature else { return }
+        // Create an alert to confirm that the user wants to update the geometry.
+        let alert = UIAlertController(title: "Confirm update", message: "Would you like to update the geometry?", preferredStyle: .alert)
+        // Clear the selection and selected feature if "No" is selected.
+        alert.addAction(UIAlertAction(title: "No", style: .cancel) { _ in
+            self.clearSelection()
+            self.selectedFeature = nil
+        })
+        alert.addAction(UIAlertAction(title: "Yes", style: .default) { _ in
+            // Set the selected feature's geometry to the new map point.
+            selectedFeature.geometry = mapPoint
+            // Update the selected feature's feature table.
+            selectedFeature.featureTable?.update(selectedFeature) { (error: Error?) in
+                if let error = error {
+                    self.presentAlert(error: error)
+                } else {
+                    // Clear selection of polyline.
+                    self.clearSelection()
+                    self.selectedFeature = nil
+                }
+            }
+        })
+        // Present the alert.
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+    // Move the last of the vertex point of the currently selected polyline to the given map point, by updating the selected feature's geometry and feature table.
+    func movePolylineVertex(mapPoint: AGSPoint) {
+        guard let selectedFeature = self.selectedFeature else { return }
+        // Create an alert to confirm that the user wants to update the geometry.
+        let alert = UIAlertController(title: "Confirm update", message: "Would you like to update the geometry?", preferredStyle: .alert)
+        // Clear the selection and selected feature if "No" is selected.
+        alert.addAction(UIAlertAction(title: "No", style: .cancel) { _ in
+            self.clearSelection()
+            self.selectedFeatureIsPolyline = false
+            self.selectedFeature = nil
+        })
+        alert.addAction(UIAlertAction(title: "Yes", style: .default) { _ in
+            // Get the selected feature's geometry as a polyline nearest to the map point.
+            guard let polyline = selectedFeature.geometry as? AGSPolyline else { return }
+            // Get the nearest vertex to the map point on the selected feature polyline.
+            guard let nearestVertex = AGSGeometryEngine.nearestVertex(in: polyline, to:
+                (AGSGeometryEngine.projectGeometry(mapPoint, to: (polyline.spatialReference)!) as? AGSPoint)!) else { return }
```

Comment:
```suggestion
            guard let polyline = selectedFeature.geometry as? AGSPolyline,
                let selectedMapPoint = AGSGeometryEngine.projectGeometry(mapPoint, to: polyline.spatialReference!) as? AGSPoint,
                let nearestVertex = AGSGeometryEngine.nearestVertex(in: polyline, to: selectedMapPoint) else { return }
```
I never see `AGSGeometryEngine` geometry fail, so it might be safe to force unwrap everywhere. If to use guard, you can combine them into 1 guard statement.